File: 백준/Silver/1735. 분수 합/분수 합.py
def GCD(a, b):
    while b > 0:
        a, b = b, a % b
    return a

A1, B1 = map(int, input().split())
A2, B2 = map(int, input().split())

son = A1 * B2 + B1 * A2
mom = B1 * B2

gcd = GCD(son, mom)

print(son//gcd, mom//gcd)

File: 백준/Silver/1735. 분수 합/분수 합.java
import java.io.*;
import java.util.*;

public class Main {
    private static long GCD(long A, long B) {
        long temp = 0;
        while (B > 0) {
            temp = A;
            A = B;
            B = temp % B;
        }
        return A;
    }

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        long A1 = sc.nextLong();
        long B1 = sc.nextLong();
        long A2 = sc.nextLong();
        long B2 = sc.nextLong();
        sc.close();
        long son = A1 * B2 + B1 * A2;
        long mom = B1 * B2;
        long gcd = GCD(son, mom);
        System.out.println(son / gcd + " " + mom / gcd);
    }
}
